packages/rollup-plugin/src/index.js

```diff
@@ -44,6 +45,15 @@ export type PluginOptions = $ReadOnly<{
   ...
 }>;
 
+function replaceFileName(original: string) {
+  const hash = crypto
+    .createHash('sha256')
+    .update(crypto.randomBytes(16))
+    .digest('hex')
+    .slice(0, 8);
+  return original.replace(/\[hash\]/g, hash);
+}
```

Comment:
```suggestion
function replaceFileName(original: string) {
  if (!original.includes('[hash]')) {
    return original;
  }
  const hash = crypto
    .createHash('sha256')
    .update(crypto.randomBytes(16))
    .digest('hex')
    .slice(0, 8);
  return original.replace(/\[hash\]/g, hash);
}
```

Reply:
Also, shouldn't the hash be hashing the contents of the generated file?

Reply:
No, in my opinion that feature request is on hashing the generated file name not the contents of the generated file. By the way following rollup document, we can use `assetFileNames` do this thing but it need to split chunk by themselves. (But we need change the emit logic)

Reply:
from everything I've read the point of putting a hash in the file name is for a cache breaker, which should likely be based on the file contents.

Reply:
My bad, now i update the code :)